source/app-development/tutorials-interactive-apps/add-custom-queue/local-dynamic-list.rst

```diff
@@ -0,0 +1,240 @@
+   defined above.
+
+   .. note::
+
+      An attribute with the field ``widget: select`` expects an ``options:``
+      field with an array of pairs. The first string in the pair is the option
+      text and the second string in the pair is the option value.
+
+      For example:
+
+      .. code-block:: yaml
+
+         widget: select
+         options:
+           - [ "Volvo", "volvo" ]
+           - [ "Ford", "ford" ]
+           - [ "Toyota", "toyota" ]
+
+      The user will see a list of options: "Volvo", "Ford", and "Toyota" to
+      choose from in the HTML form, but the backend will process a value of
+      either "volvo", "ford", or "toyota" depending on what the user chose.
+
+.. _app-development-tutorials-interactive-apps-add-custom-queue-local-dynamic-list-handle-custom-attribute:
+
+Handle Custom Attribute in Job Submission
+-----------------------------------------
+
+Now that we have our custom form attribute called ``custom_queue``, we need to
+tell our app how to handle it when submitting the job. As of right now our app
+has no idea what to do with this value when the user clicks "Launch" after
+filling out the form.
+
+.. note::
+
+   You can read more about customizing submission arguments in the
+   ``submit.yml.erb`` file for Interactive Apps under the
+   :ref:`app-development-interactive-submit` section.
+
+#. We first start with the default ``submit.yml.erb`` for the Jupyter
+   Interactive App:
+
+   .. code-block:: yaml
+
+      # ~/ondemand/dev/jupyter/submit.yml.erb
+      ---
+      batch_connect:
+        template: "basic"
+
+#. We now create a ``script:`` section if it doesn't already exist and handle
+   the value of the ``custom_queue`` attribute submitted by the user:
+
+   .. code-block:: yaml
+      :emphasize-lines: 5-
+
+      # ~/ondemand/dev/jupyter/submit.yml.erb
+      ---
+      batch_connect:
+        template: "basic"
+      script:
+        queue_name: <%= custom_queue.blank? ? "null" : custom_queue.strip %>
+
+   As we allow the user to input the value for ``custom_queue`` we need to be
+   careful when handling it:
+
+   - if it is blank we set it to the YAML value ``null``, which won't set a
+     queue when submitting the job (you can replace this with a default queue
+     for all users if you prefer)
+   - otherwise we set the queue to the user defined string with leading and
+     trailing whitespace removed
+
+   We also take advantage of the generic `OodCore::Job::Script#queue_name
+   <queue_name>`_ method to supply a queue/partition that is resource manager
+   (e.g., Slurm, Torque, ...) agnostic.
+
+   .. note::
+
+      For the queue/partition we do not need to use the ``native:`` field which
+      **must be** customized for the specific resource manager you are
+      leveraging.
+
+      You can find a list of generic fields that are resource manager agnostic
+      under the `OodCore::Job::Script <script>`_ documentation.
```

Comment:
This was a broken link and took me to https://osc.github.io/ood-documentation-test/add-queue-dropdown/app-development/tutorials-interactive-apps/add-custom-queue/script